library/firewall_lib.py

```diff
@@ -914,6 +947,19 @@ def exception_handler(exception_message):
     changed = False
     need_reload = False
 
+    # firewalld.conf
+    if firewalld_conf:
+        fw_config = fw.config()
+        if firewalld_conf.get("allow_zone_drifting") != fw_config.get_property(
+            "AllowZoneDrifting"
+        ):
+            if not module.check_mode:
+                fw_config.set_property(
+                    "AllowZoneDrifting", firewalld_conf.get("allow_zone_drifting")
+                )
+            changed = True
+            need_reload = True
```

Comment:
`fw_config.get_property("AllowZoneDrifting")` will always return "no" on versions where AllowZoneDrifting is deprecated, and setting will do nothing in this case. This causes an idempotency issue when AllowZoneDrifting is deprecated and `firewall_conf.get("allow_zone_drifting") == "yes"` by setting `changed=true` when nothing has changed, as well as prompting a reload when none is needed.

Planned fix:
Check if firewalld version >= 1.0.0, then the module will remove firewalld_conf["allow_zone_drifting"], warning the user. If this causes an error at time of arg parse (ie not enough args supplied), the module will exit early.

Does this sound like a sufficient fix?

Reply:
> Does this sound like an ideal fix?

Yes, sounds good.